### ISSUE TITLE
Gradient Accumulationの実装

### ISSUE THREAD

**Issue by lovelovetrb** [lovelovetrb/xlstm-lm#34]
イメージは以下の通り
``` python
# PyTorchでの実装例
model.zero_grad()  # 勾配をゼロに初期化
accumulation_steps = 4  # 勾配を累積するステップ数

for i in range(accumulation_steps):
    outputs = model(inputs[i])
    loss = criterion(outputs, targets[i])
    # loss値を累積ステップ数で割って調整
    loss = loss / accumulation_steps
    loss.backward()  # 勾配を計算・累積

# 累積した勾配を使って一度だけ更新
optimizer.step()
```
